allspice/utils/list_components.py

```diff
@@ -815,3 +950,74 @@ def _filter_blank_components(
         final_components.append(component)
 
     return final_components
+
+
+def _find_device_sheet(
+    device_sheet: str,
+    project_repository: Repository,
+    project_file_path: str,
+    design_reuse_repos: list[Repository],
+    logger: Logger,
+) -> tuple[Repository, pathlib.PurePosixPath]:
+    """
+    Find a device sheet by name in the design reuse repositories.
+
+    This currently does not use the directory structure of the device sheet
+    path from the PrjPcb file, and instead only matches on the filename. If
+    multiple matches are found, they are logged.
+
+    :param design_reuse_repos: The design reuse repositories to search.
+    :param device_sheet: The path of the device sheet as given in the PrjPcb.
+    :returns: The first matching repository and the path to the device sheet in
+        that repository.
+    """
+
+    matches: list[tuple[Repository, pathlib.PurePosixPath]] = []
+
+    # Paths stored in the PrjPcb are Windows paths
+    device_sheet_path = pathlib.PureWindowsPath(device_sheet)
+    # Note that this will include the .SchDoc extension.
+    device_sheet_name = device_sheet_path.name
+
+    # First, we'll resolve the actual path of the device sheet in the project
+    # repo, and if that's a real file then we use that.
+    device_sheet_path_from_repo_root = _resolve_prjpcb_relative_path(
+        device_sheet,
+        project_file_path,
+    )
+    project_repo_tree = _fetch_all_files_in_repo(project_repository)
+    if device_sheet_path_from_repo_root in project_repo_tree:
```

Comment:
Since these are windows paths I expect this check should also not be case-sensitive. That is, I expect as user could use lower-case extensions and Altium will still correctly reference that file if it is mixed-case. Up to you if we try to handle that initially.